modules/nf-core/odgi/layout/main.nf

```diff
@@ -0,0 +1,35 @@
+process ODGI_LAYOUT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::odgi=0.8.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/odgi:0.8.0--py310hc8f18ef_0':
+        'quay.io/biocontainers/odgi:0.8.0--py310hc8f18ef_0' }"
+
+    input:
+    tuple val(meta), path(graph)
+
+    output:
+    tuple val(meta), path("*.lay"), optional: true, emit: lay
+    tuple val(meta), path("*.tsv"), optional: true, emit: tsv
+    path "versions.yml"                           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    odgi \\
+        layout \\
+        --threads $task.cpus \\
+        --idx ${graph} \\
+        $args
```

Comment:
```suggestion
        --out ${prefix}.lay \\
        --tsv ${prefix}.tsv \\
        $args
```

Reply:
But the output is optional. It could the module should only output a TSV and now LAY.
Creating the files directly in the module would mean we always create both files.

Reply:
That's okay, other modules do this too. It's up to the pipeline developer to choose what to save :)

Reply:
To give you a concrete example:

I have a pangenome graph and want to visualize it in 2D. So I do `odgi layout` with LAY output. Then I use `odgi draw` and have a 2D PNG of the graph.
Or I want to interactively explore the graph with e.g. https://github.com/chfi/gfaestus. Then I would only require the TSV output of `odgi layout`.

It is really rare I want both. We want to save disk space and IO here.

Reply:
Yeah I understand what you mean! Maybe ask a second opinion on this, but I've always done it this way :)